src/core/server/http/http_tools.test.ts

```diff
@@ -57,3 +64,27 @@ describe('defaultValidationErrorHandler', () => {
     }
   });
 });
+
+test('returns 408 on timeout error', async () => {
+  const router = new Router('');
+  router.get({ path: '/', validate: false }, async (req, res) => {
+    await new Promise(resolve => setTimeout(resolve, 1000));
+    return res.ok({});
+  });
+
+  const logger = loggingServiceMock.create();
+  const server = new HttpServer(logger, 'foo');
+  const { registerRouter, server: innerServer } = await server.setup({
+    socketTimeout: 1,
+    host: '127.0.0.1',
+    maxPayload: new ByteSizeValue(1024),
+    ssl: {},
+  } as HttpConfig);
+  registerRouter(router);
+
+  await server.start();
+  await supertest(innerServer.listener)
+    .get('/')
+    .expect(408);
+  await server.stop();
```

Comment:
if assertion throws server won't stop and may affect other tests https://github.com/elastic/kibana/pull/31603/files/3d9dd70e8e73d89400f4b30476f40bc2f4cea089..90dd27f2bd4dcab7fe3d842ed90d96bc6b06525f#diff-341d10e1ae4e6cbdf938224e40cfcd03R86